pkg/agent/endpoints/sds/handler.go

```diff
@@ -224,9 +224,11 @@ func (h *Handler) startCall(ctx context.Context) (int32, []*common.Selector, fun
 		return 0, nil, nil, status.Errorf(codes.Internal, "Is this a supported system? Please report this bug: %v", err)
 	}
 
-	metrics := telemetry.WithLabels(h.c.Metrics, []telemetry.Label{{Name: telemetry.SDSPID, Value: fmt.Sprint(watcher.PID())}})
+	pidStr := fmt.Sprint(watcher.PID())
```

Comment:
using `fmt.Sprint` twice (here and in telemetry label)

may also be more consistent to use `strconv` library

Reply:
also just assign a function-scope logger with the field and carry that through the rest-
```
log := h.c.Log.WithField(telemetry.SDSPID, pidStr)
...
log.Debug("Handling...")
...
log.Debug("Finishing...")
```

following this sort of pattern, a logger within a function can pick up useful field information as the logic continues, rather than re-stating all fields in every call